packages/h5wasm/src/h5wasm-api.ts

```diff
@@ -64,26 +61,8 @@ export class H5WasmApi extends DataProviderApi {
     // Ensure all filters are supported and loaded (if available)
     await this.processFilters(h5wDataset.filters);
 
-    /* h5wasm returns bigints for (u)int64 dtypes, so we use `to_array` to get numbers instead.
-     * We do this only for datasets that are supported by at least one visualization (other than `RawVis`),
-     * so for (u)int64 scalars/arrays, and for compound datasets with at least one (u)int64 field (`MatrixVis`). */
-    if (hasInt64Type(dataset)) {
-      const rawValue = h5wDataset.to_array();
-
-      // `to_array` returns nested JS arrays for nD datasets, so we need to re-flatten them
-      const value = hasArrayShape(dataset)
-        ? flattenValue(rawValue, dataset)
-        : rawValue;
-
-      return selection ? sliceValue(value, dataset, selection) : value;
-    }
-
-    if (selection) {
-      const ranges = convertSelectionToRanges(h5wDataset, selection);
-      return h5wDataset.slice(ranges);
-    }
-
-    return h5wDataset.value;
```

Comment:
These lines move to `readValueOrSlice` and are now always called regardless of the dataset.